dockstore-webservice/src/main/java/io/dockstore/webservice/core/Entry.java

```diff
@@ -287,4 +289,71 @@ public boolean checkAndSetDefaultVersion(String newDefaultVersion) {
         }
         return false;
     }
+
+    /**
+     * Given a path (A/B/C/D), splits it into parts and returns it
+     * Note that B, C, and D are all strings, however A is either a registry or sourcecontrol enum
+     *
+     * @param path
+     * @param isTool
+     * @return An array of fields used to identify an entry
```

Comment:
Return type is known to be String[], use that rather than Object[]

Reply:
The first element in the array is either a SourceControl enum or Registry enum, so I can't use String[]

Reply:
Oh, crummy.
Maybe Pair<U, String[]> where U is a template parameter.

Reply:
This is another thing that is gonna be fixed once the registry is stored as a string.